e2e/README.md

```diff
@@ -0,0 +1,315 @@
+
+## Table of Contents
+1. [How To Write Tests](#how-to-write-tests)
+   a. [Adding A New Test](#adding-a-new-test)
+   b. [Running Tests Locally](#running-tests-locally)
+   b. [Code Samples](#code-samples)
+    - [Setup](#setup)
+    - [Creating Test Users](#creating-test-users)
+    - [Waiting](#waiting)
+    - [Query Wallet Balances](#query-wallet-balances)
+    - [Broadcasting Messages](#broadcasting-messages)
+    - [Starting the Relayer](#starting-the-relayer)
+    - [Arbitrary Commands](#arbitrary-commands)
+    - [IBC Transfer](#ibc-transfer)
+2. [Test Design](#test-design)
+   a. [ibctest](#ibctest)
+   b. [CI Configuration](#ci-configuration)
+3. [Troubleshooting](#troubleshooting)
+
+
+## How To Write Tests
+
+### Adding A New Test
+
+All tests should go under the [e2e](https://github.com/cosmos/ibc-go/tree/main/e2e) directory. When adding a new test, either add a new test function
+to an existing test suite **_in the same file_**, or create a new test suite in a new file and add test functions there.
+
+New test suites should be composed of `testsuite.E2ETestSuite`. This type has lots of useful helper functionality that will
+be quite common in most tests.
+
+> Note: see [here](#how-tests-are-run) for details about these requirements.
+
+
+### Running Tests Locally
+
+Tests can be run using a Makefile target under the e2e directory. `e2e/Makefile`
+
+There are several envinronment variables that alter the behaviour of the make target.
+
+| Environment Variable      | Description | Default Value|
+| ----------- | ----------- | ----------- |
+| SIMD_IMAGE      | The image that will be used for simd       | ibc-go-simd-e2e |
+| SIMD_TAG   | The tag used for simd        | latest|
+| RLY_TAG   | The tag used for the go relayer        | main|
+
+
+> Note: when running tests locally, **no images are pushed** to the `ghcr.io/cosmos/ibc-go-simd-e2e` registry.
+The images which are used only exist on your machine.
+
+These environment variables allow us to run tests with arbitrary verions (from branches or released) of simd
+and the go relayer.
+
+Every time changes are pushed to a branch or to `main`, a new `simd` image is built and pushed [here](https://github.com/cosmos/ibc-go/pkgs/container/ibc-go-simd-e2e).
+
+
+#### Example Command:
+
+```sh
+export SIMD_IMAGE="ghcr.io/cosmos/ibc-go-simd-e2e"
+export SIMD_TAG="pr-1650"
+export RLY_TAG="v2.0.0-rc2"
+make e2e-test suite=FeeMiddlewareTestSuite test=TestMultiMsg_MsgPayPacketFeeSingleSender
```

Comment:
Are there plans to add a `make test` to test everything?

Reply:
This would be a very heavy test, I haven't actually tested running all of them at once, and my feeling is that when we are developing tests, we will be doing so one at a time. We can depend on CI for running everything.